code/datums/status_effects/xeno_buffs.dm

```diff
@@ -156,37 +153,34 @@
 	buff_target.apply_status_effect(STATUS_EFFECT_RESIN_JELLY_COATING)
 
 /// Shares heals with the linked xeno.
-/datum/status_effect/stacking/essence_link/proc/share_heal(datum/source, amount)
+/datum/status_effect/stacking/essence_link/proc/share_heal(datum/source, amount, amount_mod, passive)
 	SIGNAL_HANDLER
-	var/mob/living/carbon/xenomorph/heal_owner
 	var/mob/living/carbon/xenomorph/heal_target
 
-	// Prevents passive regeneration from triggering this. Most values do not go past 60.
-	if(amount > -60)
-		return
-	// Prevents duplicate heals. Some heals restore both damage types, and we also have 2 signals.
-	if(heal_count)
-		heal_count--
-		return
-	heal_count++
 	if(source == link_target)
-		heal_owner = link_target
 		heal_target = link_owner
 	else
-		heal_owner = link_owner
 		heal_target = link_target
 
-	new /obj/effect/temp_visual/telekinesis(get_turf(heal_target))
-	var/heal_amount = round(clamp(amount * (DRONE_ESSENCE_LINK_SHARED_HEAL * stacks), -heal_target.maxHealth, 0))
-	var/heal_remainder = round(max(0, heal_amount - heal_target.getBruteLoss()))
-	heal_target.adjustBruteLoss(heal_amount)
-	heal_target.adjustFireLoss(-heal_remainder)
-	heal_target.adjust_sunder(heal_amount/20)
-	if(heal_amount < 0 || heal_remainder < 0)
-		heal_target.visible_message(span_xenowarning("[heal_target]'s wounds are mended by faint energies."), \
-			span_xenonotice("Through the essence link, [heal_owner] has shared [abs(heal_amount + heal_remainder)] health restoration."))
-
-// Toggles the link signals on or off.
+	// Prevents actual damage, decimals of 0, and passive healing.
+	if(amount > -1 || passive)
+		return
+
+	// Prevents duplicate heals. Some heals restore 2 damage types at once, and we also have 2 signals.
+	if(shared_heal_count)
+		shared_heal_count--
+		return
+	shared_heal_count++
+
+	new /obj/effect/temp_visual/healing(get_turf(heal_target))
+	var/heal_amount = clamp(abs(amount) * (DRONE_ESSENCE_LINK_SHARED_HEAL * stacks), 0, heal_target.maxHealth)
+	var/heal_remainder = max(0, heal_amount - heal_target.getBruteLoss())
+	heal_target.adjustBruteLoss(-heal_amount, passive = TRUE)
+	heal_target.adjustFireLoss(-heal_remainder, passive = TRUE)
+	heal_target.adjust_sunder(-heal_amount/20)
```

Comment:
Not sure I understand the sunder heal

Reply:
Can remove it if you'd like. Not much I can explain otherwise.